### PR TITLE
Parachain Template: Prune `AuxStore` bound and corresponding crate dependency

### PR DESCRIPTION
**Pull request by JoshOrndorff**
This small PR removes an unnecessary trait bound to the `AuxStore` trait from the Parachain template's `rpc.rs` file.

With that bound removed, the entire dependency on `sc-client-api` can also be removed.

### CODE REVIEW

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/4314810

**Review by bkchr — APPROVED** [review]

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/4322723

**Review by michalkucharczyk — APPROVED** [review]

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/4451615

File: cumulus/parachain-template/node/Cargo.toml [hunk 1]
```diff
@@ -29,7 +29,6 @@ pallet-transaction-payment-rpc = { path = "../../../substrate/frame/transaction-
 sc-basic-authorship = { path = "../../../substrate/client/basic-authorship" }
 sc-chain-spec = { path = "../../../substrate/client/chain-spec" }
 sc-cli = { path = "../../../substrate/client/cli" }
-sc-client-api = { path = "../../../substrate/client/api" }
```
Comment (bkchr) [inline cumulus/parachain-template/node/Cargo.toml:32]: Removing the crate was not correct, please revert.

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/4547135

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: check-tracing
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/4547198

**Pull request merged**